src/core/server/services/comments/pipeline/wordList.ts

```diff
@@ -97,19 +115,25 @@ export class WordList {
     timeout: number,
     testString: string,
     cache = true
-  ): boolean | null {
+  ): MatchResult {
     const test = this.lists(options, cache, timeout)[listName];
     if (!test) {
-      return false;
+      return {
+        isMatched: false,
+        timedOut: false,
+        matches: [],
+      };
     }
 
     const timer = createTimer();
 
+    const sanitized = this.sanitizer(testString);
+
     // Test the string against the list and timeout if it takes too long.
-    const result = test(testString);
+    const result = test(sanitized.innerHTML);
     if (result === null) {
       logger.info(
```

Comment:
While at it: I think this should be a `warning` instead.

Reply:
Is now a warning.